optuna/trial/_state.py

```diff
@@ -29,3 +29,13 @@ def __repr__(self) -> str:
     def is_finished(self) -> bool:
 
         return self != TrialState.RUNNING and self != TrialState.WAITING
+
+    def __lt__(self, state: "TrialState") -> bool:
```

Comment:
I think it's clearer to implement this as a method/function with a name. E.g.
```suggestion
    def is_promotable_to(self, state: "TrialState") -> bool:
```